src/Relude/Extra/Foldable1.hs

```diff
@@ -107,6 +107,24 @@ class Foldable f => Foldable1 f where
     minimum1 :: Ord a => f a -> a
     minimum1 = SG.getMin #. foldMap1 SG.Min
 
+    {- | The largest element of a non-empty data structure
+         with respect to the given comparison function.
+
+    >>> maximumOn1 sin (32 :| [64, 8, 128, 16])
+    8.0
```

Comment:
I think this example is a bit difficult to understand and not that helpful. I can't calculate `sin` of any number in my head, so I can't verify that this function actually works by simply looking at the example. Also, the list is specified as integral numbers but the output is written as a `Double` due to overloaded numerals. I understand that, but it makes the example not so beginner-friendly.

I propose to change example to `maximumOn1` by the `abs` function and have a list with positive and negative numbers, where the maximum by `abs` is some negative number.